lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -715,6 +712,124 @@ def test_cohort_content(self):
 
 
 @ddt.ddt
+class TestExecutiveSummaryReport(TestReportMixin, InstructorTaskCourseTestCase):
+    """
+    Tests that Executive Summary report generation works.
+    """
+    def setUp(self):
+        super(TestExecutiveSummaryReport, self).setUp()
+        self.course = CourseFactory.create()
+        CourseModeFactory.create(course_id=self.course.id, min_price=50)
+
+        self.instructor = InstructorFactory(course_key=self.course.id)
+        self.student1 = UserFactory()
+        self.student2 = UserFactory()
+        self.student1_cart = Order.get_cart_for_user(self.student1)
+        self.student2_cart = Order.get_cart_for_user(self.student2)
+
+        self.sale_invoice_1 = Invoice.objects.create(
+            total_amount=1234.32, company_name='Test1', company_contact_name='TestName',
+            company_contact_email='Test@company.com',
+            recipient_name='Testw', recipient_email='test1@test.com', customer_reference_number='2Fwe23S',
+            internal_reference="A", course_id=self.course.id, is_valid=True
+        )
+        InvoiceTransaction.objects.create(
+            invoice=self.sale_invoice_1,
+            amount=self.sale_invoice_1.total_amount,
+            status='completed',
+            created_by=self.instructor,
+            last_modified_by=self.instructor
+        )
+        self.invoice_item = CourseRegistrationCodeInvoiceItem.objects.create(
+            invoice=self.sale_invoice_1,
+            qty=10,
+            unit_price=1234.32,
+            course_id=self.course.id
+        )
+        for i in range(5):
+            coupon = Coupon(
+                code='coupon{0}'.format(i), description='test_description', course_id=self.course.id,
+                percentage_discount='{0}'.format(i), created_by=self.instructor, is_active=True,
+            )
+            coupon.save()
+
+    def test_success(self):
+        task_input = {'features': []}
+        with patch('instructor_task.tasks_helper._get_current_task'):
+            result = upload_exec_summary_report(
+                None, None, self.course.id,
+                task_input, 'generating executive summary report'
+            )
+        ReportStore.from_config(config_name='FINANCIAL_REPORTS')
+        self.assertDictContainsSubset({'attempted': 1, 'succeeded': 1, 'failed': 0}, result)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_PAID_COURSE_REGISTRATION': True})
+    def test_generate_executive_summary_report(self):
+        """
+        test to generate executive summary report
+        and then test the report authenticity.
+        """
+        self.client.login(username=self.student1.username, password='test')
```

Comment:
This bit of setup isn't directly related to the feature being tested. Move it to a helper method to keep the test method relatively short, making it easier for future developers to quickly understand the test.